tests/integration_test.rs

```diff
@@ -746,18 +748,50 @@ async fn test_detect_certificate_rotation() {
 async fn test_otel() {
     setup();
 
-    let mut otelc_config_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
-    otelc_config_path.push("tests/data/otel-collector-config.yaml");
+    let otelc_config_path =
+        PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("tests/data/otel-collector-config.yaml");
 
-    let metrics_output_file = NamedTempFile::new().unwrap();
-    let metrics_output_file_path = metrics_output_file.path();
+    let (server_ca, server_cert, server_key) = generate_tls_certs();
+
+    let server_ca_file = NamedTempFile::new().unwrap();
+    let server_cert_file = NamedTempFile::new().unwrap();
+    let server_key_file = NamedTempFile::new().unwrap();
+
+    let client_ca_file = NamedTempFile::new().unwrap();
+    let client_cert_file = NamedTempFile::new().unwrap();
+    let client_key_file = NamedTempFile::new().unwrap();
+
+    let files_and_contents = [
+        (server_ca_file.path(), &server_ca),
+        (server_cert_file.path(), &server_cert),
+        (server_key_file.path(), &server_key),
+        (client_ca_file.path(), &server_ca),
+        (client_cert_file.path(), &server_cert),
+        (client_key_file.path(), &server_key),
```

Comment:
this is wrong, you should be generating a different client cert/key for mTLS instead of reusing the server one. They can be provided by a different root CA, hence it's fine to just do something like:

```rust
let (client_ca, client_cert, client_key) = generate_tls_certs();
```